extensions/lifecycle/LifecycleQueuePopulator.js

```diff
@@ -183,6 +184,12 @@ class LifecycleQueuePopulator extends QueuePopulatorExtension {
             !(entry.key && entry.key.startsWith(mpuBucketPrefix));
     }
 
+    _isVersionedObject(mdValue) {
+        return mdValue.versionId !== undefined ||
+            mdValue.isNull ||
+            mdValue.isDeleteMarker;
```

Comment:
I think `isDeleteMarker` implies a `versionId` exists, so may not be needed to check explicitly, but in doubt ok to keep it. (it may also be the case for `isNull` as well)